lib/utils.js

```diff
@@ -9,8 +9,10 @@ exports.defineCachedRelations = defineCachedRelations;
 exports.sortObjectsByIds = sortObjectsByIds;
 exports.setScopeValuesFromWhere = setScopeValuesFromWhere;
 exports.mergeQuery = mergeQuery;
+exports.createPromiseCallback = createPromiseCallback;
 
 var traverse = require('traverse');
+var Promise = require('bluebird');
```

Comment:
Can we check if the global `Promise` exists (`typeof Promise !== undefined`) and use `bluebird` as the fallback? For example, iojs already supports `Promise` natively.